### PR TITLE
Add support for TI TMP116

### PR DESCRIPTION
**Pull request by gon1332**
Added dts/driver/sample support for [TI TMP116 temperature driver](http://www.ti.com/lit/ds/symlink/tmp116.pdf).

*NOTE 1*: This driver does not utilize the `ALERT` pin due to the fact it is tested on a custom shield connected to a `nucleo_f401re`. This shield has this pin not connected.

*NOTE 2*: I've observed that there are also two other similar drivers, TMP007 and TMP112 ported in Zephyr. So there is some code duplication and also the above drivers do not have dts bindings and use the kconfig symbols. Any ideas on what to do with those? I don't have them available to touch them.

### CODE REVIEW

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please add this sensor to `tests/drivers/build_all/sensors_i_z.conf`

File: dts/bindings/sensor/ti,tmp116.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# SPDX-License-Identifier: Apache-2.0
+#
+
+title: Texas Instruments Temperature Sensor TMP116
+
+description: >
+    This is a representation of the TI Temperature sensor TMP116
+
+inherits:
+    !include i2c-device.yaml
```
[the remark this answers is not in the document]
Comment (gon1332) [reply dts/bindings/sensor/ti,tmp116.yaml]: Done.

File: drivers/sensor/tmp116/tmp116.c [hunk 2]
```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2019 Centaur Analytics, Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <i2c.h>
+#include <sensor.h>
```
[the remark this answers is not in the document]
Comment (gon1332) [reply drivers/sensor/tmp116/tmp116.c]: Done.

File: tests/drivers/build_all/sensors_i_z.conf [hunk 3]
```diff
@@ -31,4 +31,5 @@ CONFIG_TH02=y
 CONFIG_TI_HDC=y
 CONFIG_TMP007=y
 CONFIG_TMP112=y
+CONFIG_TMP116=y
```
Comment (MaureenHelm) [inline tests/drivers/build_all/sensors_i_z.conf]: You also need to add macros to `tests/drivers/build_all/dts_fixup.h` to fix the shippable failure

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: tests/drivers/build_all/sensors_i_z.conf (hunk 3 above)
Comment (gon1332) [reply tests/drivers/build_all/sensors_i_z.conf]: Already done and waiting. Thanks!

File: tests/drivers/build_all/dts_fixup.h [hunk 4]
```diff
@@ -337,4 +337,11 @@
 #define DT_INST_0_MICROCHIP_ENC28J60_SPI_MAX_FREQUENCY	0
 #endif
 
+#ifndef DT_INST_0_TI_TMP116_LABEL
+#define DT_INST_0_TI_TMP116_LABEL ""
+#define DT_INST_0_TI_TMP116_BASE_ADDRESS 0
+#define DT_INST_0_TI_TMP116_BUS_NAME ""
+#define DT_TI_TMP116_BUS_I2C 1
+#endif
```
Comment (MaureenHelm) [inline tests/drivers/build_all/dts_fixup.h]: Move these defines to the `#if defined(CONFIG_HAS_DTS_I2C)` section. They're currently in the SPI section

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: tests/drivers/build_all/dts_fixup.h (hunk 4 above)
Comment (gon1332) [reply tests/drivers/build_all/dts_fixup.h]: Done.

**Comment by gon1332** [comment]
Made zephyr bot happy.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Review by MaureenHelm — APPROVED** [review]
@dbkinder please review

File: samples/sensor/tmp116/README.rst [hunk 5]
```diff
@@ -0,0 +1,52 @@
+.. _ti_tmp116_sample:
+
+TI_TMP116 Sample
+################
+
+Description
+***********
+
+This sample application periodically takes Temperature using the ti_tmp116
```
Comment (dbkinder) [inline samples/sensor/tmp116/README.rst]: recommend:
```
... takes temperature readings from the ti_tmp116 ...
```

File: samples/sensor/tmp116/README.rst [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+************
+
+This sample needs a TI TMP116 sensor connected to the target board's I2C
+connector.
+
+
+Wiring
+******
+
+This sample is tested with the STM32F401RE nucleo.
```
Comment (dbkinder) [inline samples/sensor/tmp116/README.rst]: recommend:
```
... with the nucleo STM32F401RE board.
```

File: samples/sensor/tmp116/README.rst [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+* Breakout **GND** pin <--> Nucleo **GND** pin
+* Breakout **VCC** pin <--> Nucleo **3V3** pin
+* Breakout **SDA** pin <--> Nucleo **CN5-D14** pin
+* Breakout **SCL** pin <--> Nucleo **CN5-D15** pin
+
+Building and Running
+********************
+
+This sample builds one application for the TMP 116 sensor.
+Build/Flash Steps:
```
Comment (dbkinder) [inline samples/sensor/tmp116/README.rst]: You should also mention connecting a USB cable between the development computer and the board (for flashing) and running a terminal emulator (e.g., minicom) and specify the settings, to see the console output.

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Review by dbkinder — APPROVED** [review]
thanks for doc updates

**Pull request merged**